
# HG changeset patch
# User Phil Thompson <phil@riverbankcomputing.com>
# Date 1656234101 -3600
# Node ID 1430b279a3c90da85f2cadd97ec833deae14f446
# Parent  6eff81a158220a6fdfda8c4b14dd4fb2ddbe3ac8
Fixed the handling of exceptions that sub-class C++ exceptions.

diff -r 6eff81a15822 -r 1430b279a3c9 NEWS
--- a/NEWS	Fri Jun 17 14:25:16 2022 +0100
+++ b/NEWS	Sun Jun 26 10:01:41 2022 +0100
@@ -1,3 +1,6 @@
+v6.6.3 26th June 2022
+  - Bug fixes.
+
 v6.6.2 6th June 2022
   - Enums created with ABI v13 now handle unexpected values by creating
     corresponding pseudo-members.
diff -r 6eff81a15822 -r 1430b279a3c9 sipbuild/generator/parser/rules.py
--- a/sipbuild/generator/parser/rules.py	Fri Jun 17 14:25:16 2022 +0100
+++ b/sipbuild/generator/parser/rules.py	Sun Jun 26 10:01:41 2022 +0100
@@ -2293,6 +2293,7 @@
 
     if len(p) == 4:
         base = p[2]
+        base.make_absolute()
 
         # See if it is a project-defined exception.
         for xd in p.parser.pm.spec.exceptions:

